include/rcutils/macros.h

```diff
@@ -142,6 +142,19 @@ extern "C"
 # define RCUTILS_UNLIKELY(x) (x)
 #endif  // _WIN32
 
+// Provide the compiler a hint about an argument being nonnull when possible.
+#ifndef _WIN32
+# define RCUTILS_NONNULL __attribute__((__nonnull__))
+# define RCUTILS_NONNULL_ARGS(...) __attribute__((__nonnull__ (__VA_ARGS__)))
+# define RCUTILS_RETURNS_NONNULL __attribute__((__returns_nonnull__))
+# define RCUTILS_HAS_NONNULL 1
```

Comment:
What's the reason for this particular macro?

Reply:
So that you can write code that differs based on whether or not the compiler supports this feature. I needed it for a pull request where I'm using these macros.

Right now the code lives here, but I might move it to something like `rcpputils` later: https://github.com/space-ros/demos/blob/e79d1a48cca84d383b3433e389799d9e5b7d3c87/space_ros_memory_allocation_demo/include/space_ros_memory_allocation_demo/memory_resource_helper.hpp#L128-L136